### ISSUE TITLE
Loot sorting fails in Fallout: NV

### ISSUE THREAD

**Issue by phen0mX** [limo-app/limo#11]
Sorting plugins in Fallout New Vegas results in:

> [Error]: yaml-cpp: error at line 19, column 9: bad conversion: 'text' key missing from 'message content' object

Even with no additional mods.

![Bildschirmfoto vom 2024-08-25 17-56-22](https://github.com/user-attachments/assets/b92b128f-7454-42b3-b366-59881ac739f8)

**Comment by phen0mX** [comment]
Good catch! I had already read that, but didn't want to bring up game specific issues.
As the whole concept of Limo seems to try and be as generic as possible.
I only mentioned FNV because it's (masterlist) is the source of the error I encountered.

**Comment by limo-app** [comment]
Can reproduce this by installing a single mod ([this one](https://www.nexusmods.com/newvegas/mods/35979?tab=files)). I will try to debug this, but it looks like an internal issue of libloot in which case i will raise this with upstream.

**Comment by phen0mX** [comment]
It only happened with a mod installed?
For me it happened with my modlist (didn't check single mods) and 
when I "undeployed" all mods using an empty profile.
Btw. is there a simpler way to "undeploy" all mods?

**Comment by phen0mX** [comment]
Applying a loadorder.txt, like is mentioned [here](https://vivanewvegas.moddinglinked.com/basefinish.htmll).
Would be a potential way to workaround the no/outdated masterlist issue.

The contents seem quite easy to parse at least:
```
# This file was automatically generated by Mod Organizer.
FalloutNV.esm
DeadMoney.esm
HonestHearts.esm
OldWorldBlues.esm
LonesomeRoad.esm
GunRunnersArsenal.esm
ClassicPack.esm
MercenaryPack.esm
TribalPack.esm
CaravanPack.esm

**Comment by limo-app** [comment]
Sorry for the confusion, i meant to say that even if you just use a single mod, the issue still occurs. So its probably not related to the mods you used but a general problem with NVs masterlist. By valid yaml is simply mean that the file contains no yaml syntax errors, it could still contain other errors.

**Comment by limo-app** [comment]
> Applying a loadorder.txt, like is mentioned [here](https://vivanewvegas.moddinglinked.com/basefinish.htmll).

This link does not work for me. The loadorder file is currently parsed and edited by Limo, so you can sort your mods manually with drag and drop. The masterlist is just needed for automatic sorting with libloot.

> Btw. is there a simpler way to "undeploy" all mods?

Currently not. I thought having a "Vanilla" profile would be sufficient. I thought about adding an "Undeploy" button but decided against it since i thought it would rare be useful under normal circumstances.

**Comment by phen0mX** [comment]
> This link does not work for me. The loadorder file is currently parsed and edited by Limo, so you can sort your mods manually with drag and drop. The masterlist is just needed for automatic sorting with libloot.
Fixed. They are different files, "plugins.txt" is what's read by the game and "loadorder.txt" is more of an instruction to the mod manager to reorder the plugins. Useful for guides, bundles stuff like that.
Because dragging around 20+ plugins gets annoying quite quickly, although it does work.

> Currently not. I thought having a "Vanilla" profile would be sufficient. I thought about adding an "Undeploy" button but decided against it since i thought it would rare be useful under normal circumstances.

I have to say the current solution does work without issue, but it's extremely unintuitive.
A fixed read-only "Vanilla" profile would be significantly better.

**Comment by limo-app** [comment]
> but it's extremely unintuitive.

That's fair. I think the best solution to this would actually be some sort of un-deploy  button after all.

> "loadorder.txt" is more of an instruction to the mod manager to reorder the plugins

Since it looks like some games, like NV, might be better of without LOOT's auto sorting, it might be best to add an option that replaces this way of sorting with a new algorithm that respects the order in that file.

**Comment by geowarin** [comment]
Hello, I ran into the same problem with Oblivion (yaml-cpp: error at line 41, column 9: bad conversion: 'test' key missing from 'message content' object)

![image](https://github.com/user-attachments/assets/4cab4e6c-a179-4ea1-9e04-62471849f98a)

if `masterlist.yaml` is the culprit it's around the same type of structure.

```yaml
  - &obsolete
    type: say
    content:
      - lang: en            <========== line 41
        str: 'Obsolete. Update to latest version.'
      - lang: ru
        str: 'Устарело. Обновите до последней версии.'
      - lang: es
        str: 'Obsoleto. Actualizar a la última versión.'
      - lang: ko
        str: '사용되지 않습니다. 최신 버전으로 업데이트하십시오.'
      - lang: zh_CN
        str: '已过时。请升级到最新版本。'
```

Another problem I noticed is that, the "Plugins" (loot) deployer writes the `plugins.txt` file, whereas Oblivion will read the `Plugins.txt` file (uppercase P)

**Comment by geowarin** [comment]
As suggested in the issue @limo-app [opened upstream](https://github.com/loot/libloot/issues/96), I downloaded the masterlist file from the `0.21` branch (which is the default branch on the repo) and the problem parsing the yaml file was gone.

I don't know of a way to download from `raw.githubusercontent` without specifying a branch so... maybe it's ok to har-code 0.21 for now?
Otherwise, we could use the [github API](https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#get-a-repository) to query the default branch but it looks like a hassle...

However, the issue with the case issue with `plugin.txt` remains. Should I open a separate issue?

**Comment by phen0mX** [comment]
> As suggested in the issue @limo-app [opened upstream](https://github.com/loot/libloot/issues/96), I downloaded the masterlist file from the `0.21` branch (which is the default branch on the repo) and the problem parsing the yaml file was gone.

Can confirm. After downloading the 0.21 masterlist LOOT works for FNV.

**Comment by limo-app** [comment]
> I don't know of a way to download from `raw.githubusercontent` without specifying a branch so... maybe it's ok to har-code 0.21 for now?

Seeing as there hasn't been much activity in that repo recently, i believe hard coding 0.21 is probably the best option right now.

> However, the issue with the case of `plugin.txt` remains. Should I open a separate issue?

Yes please. I don't own most of the bethesda games but will try to find the correct file names for all of them for the next update.

**Comment by limo-app** [comment]
[1d7129d](https://github.com/limo-app/limo/commit/1d7129de828007ef589de4f3823f510cfd70ef8b) now uses v0.21 for all masterlists.
I have also updated the naming for plugins.txt files. As far as i can see only oblivion uses the upper case version. The new implementation now respects the exact file name of any previously existing plugins.txt file. If none is found it defaults to `Plugins.txt` for Oblivion and `plugins.txt` for everything else.

**Comment by limo-app** [comment]
Both the original sorting issue and the capitalization of Plugins.txt for Oblivion are fixed in 1.0.5 which is live on Flathub. Closing
